core/src/test/scala/backbone/BackboneSpec.scala

```diff
@@ -1,30 +1,131 @@
+import backbone.format.DefaultMessageWrites
+import backbone.publisher.PublisherSettings
 import backbone.scaladsl.Backbone
 import backbone.testutil.Implicits._
-import com.amazonaws.services.sns.AmazonSNSAsyncClient
-import org.scalatest.WordSpec
+import backbone.testutil.{MockSNSAsyncClient, PublishHandler, TestActorSystem}
+import com.amazonaws.auth.{AWSStaticCredentialsProvider, BasicAWSCredentials}
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration
+import com.amazonaws.services.sns.model.PublishRequest
+import com.amazonaws.services.sqs.{AmazonSQSAsync, AmazonSQSAsyncClientBuilder}
+import org.mockito.ArgumentMatchers.{any, eq => meq}
+import org.mockito.Mockito.verify
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{fixture, MustMatchers, Outcome}
 
-class BackboneSpec extends WordSpec with DefaultTestContext with MockitoSugar with ScalaFutures {
+import scala.concurrent.Future
+import scala.concurrent.duration._
 
-  private[this] implicit val snsClient = mock[AmazonSNSAsyncClient]
+class BackboneSpec
+    extends fixture.WordSpec
+    with MockitoSugar
+    with ScalaFutures
+    with MustMatchers
+    with TestActorSystem
+    with MockSNSAsyncClient
+    with DefaultMessageWrites {
 
   "Backbone.consume" should {
+    // TODO: consume tests
 
-    "doe something" in {
+    "doe something" in { ctx =>
+      import ctx._
 
-      val settings = ConsumerSettings(List.empty, List.empty, "Queue-name")
-      val backbone = Backbone()
+      val settings = ConsumerSettings(List.empty, "Queue-name")
 
       backbone.consume[String](settings) { s =>
         Consumed
       }
+    }
+
+  }
+
+  "Backbone.publishAsync" should {
+
+    "publish a single message to an SNS topic" in { ctx =>
+      import ctx._
+
+      val result = backbone.publishAsync("message", publisherSettings)
+
+      whenReady(result) { res =>
+        res mustBe Done
+        verify(snsClient).publishAsync(meq(new PublishRequest(publisherSettings.topicArn, "message")),
+                                       any[PublishHandler])
+      }
+    }
+
+    "publish multiple messages to an SNS topic" in { ctx =>
+      import ctx._
+
+      val messages = "message-1" :: "message-2" :: Nil
+      val result   = backbone.publishAsync(messages, publisherSettings)
+
+      whenReady(result) { res =>
+        res mustBe Done
+        verify(snsClient).publishAsync(meq(new PublishRequest(publisherSettings.topicArn, "message-1")),
+                                       any[PublishHandler])
+        verify(snsClient).publishAsync(meq(new PublishRequest(publisherSettings.topicArn, "message-2")),
+                                       any[PublishHandler])
+      }
+    }
+
+    "publish messages from an ActorRef" in { ctx =>
+      import ctx._
 
+      val actorRef = backbone.actorPublisher[String](publisherSettings)
+
+      within(1.seconds) {
+        actorRef ! "message-1"
```

Comment:
Don't place the message sending inside the within because it will resend the messages then.